### PR TITLE
Add audio-video-player sample

### PR DESCRIPTION
**Pull request by chcunningham**
Live demo here: https://webcodecs-samples.netlify.app/audio-video-player/audio_video_player.html

Authored in collaboration w/ @padenot 

Originally hosted here:
https://github.com/chcunningham/wc-talk
    
This is mostly a copy/paste of that, but I've moved demuxing and
decoding to a dedicated "media worker" and tried to clean things up a
bit.

### CODE REVIEW

**Comment by chcunningham** [comment]
@sandersdan FYI - this sample has the "pull demuxer" wrapper. I intend to make an interface for this in a follow up CL (and use it for the ffmpeg demuxer)

File: samples/audio-video-player/mp4_pull_demuxer.js [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+// Wrapper around MP4Box.js that shims pull-based demuxing on top their
+// push-based API.
+export class MP4PullDemuxer {
+  constructor(uri) {
+    this.source = new MP4Source(uri);
+    this.readySamples = [];
+    this._pending_read_resolver = null;
+  }
+
+  getAvcDescription(avccBox) {
```
Comment (sandersdan) [inline samples/audio-video-player/mp4_pull_demuxer.js:19]: Nit: See my webgpu branch for a simpler approach (https://github.com/sandersdan/webcodecs/blob/webgpu/samples/mp4-decode/mp4_demuxer.js).

File: samples/audio-video-player/server.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      "js": "text/javascript",
+      "wasm": "application/wasm",
+      "css": "text/css"
+    };
+
+const serverPort = parseInt(port, 10);
+
+http.createServer({}, function(request, response) {
+
+  // This server is insecure. Only listen for conncections on localhost.
```
Comment (sandersdan) [inline samples/audio-video-player/server.js:20]: connections

File: samples/audio-video-player/server.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      console.log("serving " + filename);
+
+      response.writeHead(200, { "Content-Type": mimeType ,
+        "Cross-Origin-Opener-Policy": "same-origin",
+        "Cross-Origin-Embedder-Policy": "require-corp" });
+      response.write(file, "binary");
+      response.end();
+    });
+  });
+}).listen(serverPort);
```
Comment (sandersdan) [inline samples/audio-video-player/server.js:70]: Note: If you listen on hostname "127.0.0.1" or "localhost" then you shouldn't have to filter requests.

File: samples/audio-video-player/audio_renderer.js [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+    this.interleavingBuffers = [];
+
+    this.init_resolver = null;
+    let promise = new Promise(resolver => (this.init_resolver = resolver));
+
+    this.fillDataBuffer();
+    return promise;
+  }
+
+  setVolume(volume) {
```
Comment (tguilbert-google) [inline samples/audio-video-player/audio_renderer.js:63]: Is this ever used? I only see the WebAudioController handling volume.

**Review by tguilbert-google — APPROVED** [review]
LGTM in general!

File: samples/audio-video-player/mp4_pull_demuxer.js (hunk 1 above)
Comment (chcunningham) [reply samples/audio-video-player/mp4_pull_demuxer.js:19]: Done!

File: samples/audio-video-player/audiosink.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+registerProcessor("AudioSink", class AudioSink extends AudioWorkletProcessor {
+  constructor(options) {
+    super();
+    let sab = options.processorOptions.sab;
+    this.consumerSide = new RingBuffer(sab, Float32Array);
+    this.mediaChannelCount = options.processorOptions.mediaChannelCount;
+    this.deinterleaveBuffer = new Float32Array(this.mediaChannelCount * 128);
```
[the remark this answers is not in the document]
Comment (chcunningham) [reply samples/audio-video-player/audiosink.js]: https://www.w3.org/TR/webaudio/#render-quantum-size

I added a const and comment for better documentation.

**Review by chcunningham — COMMENTED** [review]
Thanks!

**Comment by chcunningham** [comment]
@padenot @aboba - FYI, merging. Later feedback welcome. Getting this in now so I can build on top of it.

**Pull request merged**